src/wp-includes/rest-api/endpoints/class-wp-rest-block-types-controller.php

```diff
@@ -0,0 +1,552 @@
+	 * @return WP_Error|bool True if the request has read access for the item, WP_Error object otherwise.
+	 */
+	public function get_item_permissions_check( $request ) {
+		$check = $this->check_read_permission();
+		if ( is_wp_error( $check ) ) {
+			return $check;
+		}
+		$block_name = sprintf( '%s/%s', $request['namespace'], $request['name'] );
+		$block_type = $this->get_block( $block_name );
+		if ( is_wp_error( $block_type ) ) {
+			return $block_type;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Checks whether a given block type should be visible.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @return WP_Error|bool True if the block type is visible, otherwise false.
+	 */
+	protected function check_read_permission() {
+		if ( current_user_can( 'edit_posts' ) ) {
+			return true;
+		}
+		foreach ( get_post_types( array( 'show_in_rest' => true ), 'objects' ) as $post_type ) {
+			if ( current_user_can( $post_type->cap->edit_posts ) ) {
+				return true;
+			}
+		}
+
+		return new WP_Error( 'rest_block_type_cannot_view', __( 'Sorry, you are not allowed to manage block types.' ), array( 'status' => rest_authorization_required_code() ) );
+	}
+
+	/**
+	 * Get the block, if the name is valid.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param string $name Block name.
+	 * @return WP_Block_Type|WP_Error Block type object if name is valid, WP_Error otherwise.
+	 */
+	protected function get_block( $name ) {
+		$block_type = $this->block_registry->get_registered( $name );
+		if ( empty( $block_type ) ) {
+			return new WP_Error( 'rest_block_type_invalid', __( 'Invalid block type.' ), array( 'status' => 404 ) );
+		}
+
+		return $block_type;
+	}
+
+	/**
+	 * Retrieves a specific block type.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|WP_REST_Response Response object on success, or WP_Error object on failure.
+	 */
+	public function get_item( $request ) {
+		$block_name = sprintf( '%s/%s', $request['namespace'], $request['name'] );
+		$block_type = $this->get_block( $block_name );
+		if ( is_wp_error( $block_type ) ) {
+			return $block_type;
+		}
+		$data = $this->prepare_item_for_response( $block_type, $request );
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Prepares a block type object for serialization.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param WP_Block_Type   $block_type Block type data.
+	 * @param WP_REST_Request $request    Full details about the request.
+	 * @return WP_REST_Response block type data.
+	 */
+	public function prepare_item_for_response( $block_type, $request ) {
+
+		$fields = $this->get_fields_for_response( $request );
+		$data   = array();
+
+		if ( rest_is_field_included( 'attributes', $fields ) ) {
+			$data['attributes'] = $block_type->get_attributes();
+		}
+
+		if ( rest_is_field_included( 'is_dynamic', $fields ) ) {
+			$data['is_dynamic'] = $block_type->is_dynamic();
+		}
+
+		$schema       = $this->get_item_schema();
+		$extra_fields = array(
```

Comment:
Can we simplify `$extra_fields` to an array of values? We fixed all differences and we use the same names in both REST API and `WP_Block_Type` class.

Reply:
Cc @spacedmonkey.

Reply:
Fixed in https://github.com/WordPress/wordpress-develop/pull/317/commits/a39c6fcf21960ae3b77b6c2a615abfbffa281015